### PR TITLE
[15.0][IMP] commission: multi-currency support

### PR DESCRIPTION
**Pull request by alexeirivera87**
Hello

Could someone please check these changes ?

regards

### CODE REVIEW

**Comment by rousseldenis** [comment]
@alexeirivera87 Thanks for this. This deserves some tests. Could you add them ?

**Comment by alexeirivera87** [comment]
@rousseldenis  Thanks. I added some tests. Let me know if it is correct now. Regards

**Comment by alexeirivera87** [comment]
Hello 

Any news on this PR ?

@rousseldenis 

@pedrobaeza 

regards

**Comment by alexeirivera87** [comment]
@ernestotejeda

**Review by AungKoKoLin1997 — APPROVED** [review]
Code review and function test: LGTM

File: account_commission/models/account_move.py [hunk 1]
```diff
@@ -199,6 +199,7 @@ class AccountInvoiceLineAgent(models.Model):
     currency_id = fields.Many2one(
         related="object_id.currency_id",
         readonly=True,
+        store=True,
```
Comment (pedrobaeza) [inline account_commission/models/account_move.py:202]: This will be fatal on module upgrade, needing to store it for all the existing lines. Why do you need this?

File: account_commission/models/commission_settlement.py [hunk 2]
```diff
@@ -78,15 +78,20 @@ def action_invoice(self):
     def _get_invoice_partner(self):
         return self[0].agent_id
 
+    def _get_invoice_currency(self):
+        return self[0].currency_id
```
Comment (pedrobaeza) [inline account_commission/models/commission_settlement.py:82]: Why is this method needed?

File: account_commission/wizards/commission_make_settle.py [hunk 3]
```diff
@@ -14,16 +14,19 @@ class CommissionMakeSettle(models.TransientModel):
         ondelete={"sale_invoice": "cascade"},
     )
 
+    def _get_account_settle_domain(self, agent, date_to_agent):
```
Comment (pedrobaeza) [inline account_commission/wizards/commission_make_settle.py:17]: This is outside the scope of the PR. I'm not against it. Just that is should be done apart.

File: commission/wizards/commission_make_settle.py [hunk 4]
```diff
@@ -66,26 +67,28 @@ def _get_next_period_date(self, agent, current_date):
...
                 ("company_id", "=", company.id),
+                ("currency_id", "=", currency.id),
                 ("state", "=", "settled"),
                 ("settlement_type", "=", self.settlement_type),
             ],
             limit=1,
         )
 
-    def _prepare_settlement_vals(self, agent, company, sett_from, sett_to):
+    def _prepare_settlement_vals(self, agent, company, currency, sett_from, sett_to):
```
Comment (pedrobaeza) [inline commission/wizards/commission_make_settle.py:85]: Changing a signature in a stable version is not correct, as it will convert in incompatible all the existing overwrites of this method. Don't you have any other way?

**Comment by jguenat** [comment]
Hello @alexeirivera87 
Do you have time to finish this PR? Otherwise I could continue.
Thanks

**Comment by pedrobaeza** [comment]
Superseded by #446
